components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -32,8 +33,17 @@ Nudging::Nudging (const ekat::Comm& comm, const ekat::ParameterList& params)
     m_src_pres_type = STATIC_1D_VERTICAL_PROFILE;
     // Check for a designated source pressure file, default to first nudging data source if not given.
     m_static_vertical_pressure_file = m_params.get<std::string>("source_pressure_file",m_datafiles[0]);
+  } else if (src_pres_type=="TIME_DEPENDENT_3D_HYBRID") {
+    m_src_pres_type = TIME_DEPENDENT_3D_HYBRID;
+    int first_file_levs = scorpio::get_dimlen(m_datafiles[0],"lev");
```

Comment:
I think this check should be done regardless of `src_pres_type`, no? If there are 2+ files, we should always verify that their dimensions match.

Reply:
you mean for all time-varying src_pres_type (not including STATIC_1D_VERTICAL_PROFILE) right?

Reply:
Well, also for static, no? What if the user specifies two files, and accidentally they are from different runs?